src/main/java/org/example/user/FileIdGenerator.java

```diff
@@ -0,0 +1,49 @@
+package org.example.user;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Primary;
+import org.springframework.stereotype.Component;
+
+import java.io.*;
+
+@Component
+@Primary
+public class FileIdGenerator implements UserIdGenerator {
+
+    private int id;
+    private final String filePath;
+
+    public FileIdGenerator(@Value("${file.path.id}") String filePath) {
+        this.id = loadIdFromFile(filePath);
+        this.filePath = filePath;
+    }
+
+    private void saveIdToFile(int id) throws IOException {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(filePath))) {
+            writer.write(String.valueOf(id));
+        }
+    }
+
+    private int loadIdFromFile(String filePath) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
+            String line = reader.readLine();
+            if (line != null) {
+                id = Integer.parseInt(line);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
we continue to work when failed to read from file on FileIdGenerator creation time. 
Just imagine that filePath is some wrong pass and file is not exists. I'd like to know about it on application start up. Not when we try to save first user